org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -383,7 +396,6 @@ public void didChangeConfiguration(DidChangeConfigurationParams params) {
 		} catch (CoreException e) {
 			JavaLanguageServerPlugin.logException(e.getMessage(), e);
 		}
-		pm.registerWatchers();
```

Comment:
why did we register watchers on config change in the 1st place?

Reply:
We don't do that anymore.
Now, we register watchers after initialization of the server - https://github.com/eclipse/eclipse.jdt.ls/pull/985/commits/e403e55a8dd1a51d143cede37ec424b9cad2297b#diff-836a9345804342fc811e247515539e98R242